src/realm_dart_realm.h

```diff
@@ -0,0 +1,39 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_DART_REALM_H
+#define REALM_DART_REALM_H
+
+#include "realm.h"
+#include "dart_api_dl.h"
+
+typedef int (*realm_dart_data_initialization_func_t)(Dart_Handle callback_func, realm_t*);
```

Comment:
Wish I could declare that as a boolean return type but that generates Uin8 in the ffi bindings and I couldn't figure out the dart signature that would be compatible with it.

Reply:
When you do `Pointer.fromFunction` the argument can just be a Dart function returning bool. It should work 🤔